verification/curator-service/ui/src/components/SourceTable.tsx

```diff
@@ -77,14 +94,40 @@ export default class SourceTable extends React.Component<{}, SourceTableState> {
         })
     }
 
+    createSourceFromRowData(rowData: TableRow) {
+        return {
+            _id: rowData._id,
+            name: rowData.name,
+            origin: {
+                url: rowData.url
+            }
+        }
+    }
+
+    validateRequired(field: String) {
+        return field?.trim() !== "";
+    }
+
     render() {
         return (
             <Paper>
                 <MaterialTable
                     columns={[
                         { title: 'ID', field: '_id', editable: "never" },
                         { title: 'Name', field: 'name' },
```

Comment:
In the addSource method above it requires name, so we should add the editComponent here as well so that the UI indicates that the field is required.

Reply:
SG, updated.